telegram/ext/_application.py

```diff
@@ -1672,3 +1688,64 @@ async def process_error(
 
         _logger.exception("No error handlers are registered, logging exception.", exc_info=error)
         return False
+
+
+class BaseProcessor:
```

Comment:
let's move this to a standalone file :)
`SimpleProcessor` could maybe be put in the same file … I guess Pool would like that :D

the two classes also need docstrings